compiler/rustc_lint/src/builtin.rs

```diff
@@ -2890,22 +2890,55 @@ impl EarlyLintPass for SpecialModuleName {
     }
 }
 
-pub use rustc_session::lint::builtin::UNEXPECTED_CFGS;
+declare_lint! {
+    /// The `unexpected_cli_cfgs` lint detects unexpected conditional compilation given
+    /// in the command line.
+    ///
+    /// ### Example
+    ///
+    /// ```text
+    /// rustc --cfg 'foo' --check-cfg 'cfg()'
+    /// ```
+    ///
+    /// ```rust,ignore (needs command line option)
+    /// fn main() {}
+    /// ```
+    ///
+    /// This will produce:
+    ///
+    /// ```text
+    /// warning: unexpected `foo` as condition name
+    ///    |
+    ///    = help: was set with `--cfg` but isn't in the `--check-cfg` expected names
+    /// ```
+    ///
+    /// ### Explanation
+    ///
+    /// This lint is only active when a `--check-cfg='cfg(...)'` option has been passed
+    /// to the compiler and triggers whenever an unknown condition name or value is used
+    /// on `--cfg` command line arguments.
+    pub UNEXPECTED_CLI_CFGS,
+    Allow,
+    "detects unexpected names and values in `--cfg`",
+}
 
-declare_lint_pass!(UnexpectedCfgs => [UNEXPECTED_CFGS]);
+declare_lint_pass!(UnexpectedCliCfgs => [UNEXPECTED_CLI_CFGS]);
 
-impl EarlyLintPass for UnexpectedCfgs {
+impl EarlyLintPass for UnexpectedCliCfgs {
     fn check_crate(&mut self, cx: &EarlyContext<'_>, _: &ast::Crate) {
         let cfg = &cx.sess().parse_sess.config;
         let check_cfg = &cx.sess().parse_sess.check_config;
         for &(name, value) in cfg {
             match check_cfg.expecteds.get(&name) {
                 Some(ExpectedValues::Some(values)) if !values.contains(&value) => {
                     let value = value.unwrap_or(kw::Empty);
-                    cx.emit_lint(UNEXPECTED_CFGS, BuiltinUnexpectedCliConfigValue { name, value });
+                    cx.emit_lint(
+                        UNEXPECTED_CLI_CFGS,
```

Comment:
This isn't technically required, but letting it under the `unexpected_cfgs` lint is weird as it would mean that `unexpected_cli_cfgs` would actually mean `unexpected_cli_name_cfgs`, which is even weirder.